src/popup/components/images.tsx

```diff
@@ -48,11 +48,22 @@ export function ImageThumbnail({ image, startEditingImage, clickDeleteImage, del
   )
 }
 
+function ImageSpinner() {
+  return (
+    <div className="image-spinner">
+      <div className="spinner-container">
+        <div className="spinner" />
+      </div>
+    </div>
+  )
+}
+
 export function Images({ feedback, startEditingImage, clickDeleteImage, deleteImageDisabled }: ImagesProps): null | JSX.Element {
-  if (!feedback.images.length) return null
+  if (!feedback.images.length && !feedback.addingImages) return null
 
   return (
     <div className="images">
+      {!!feedback.addingImages && <ImageSpinner />}
       {feedback.images.map((image, index) => (
```

Comment:
How do we know not to post a tweet until all images were added? How do we handle the situation when a user wants to abort the process? I think we should still have an 'X' option

Reply:
Great call, adding that